lib/rouge/lexer.rb

```diff
@@ -13,8 +13,15 @@ class Lexer
     include Token::Tokens
 
     @option_docs = {}
+    @registry = {}
+    @lazy_constants = {}
```

Comment:
Forgive me, it's rather late here and I'm not sure I'm parsing this properly but this `@registry` is an instance variable defined on the class `Lexer`, right? That's distinct from the '@registry` that's references in `Lexer.registry` which is an instance variable on the singleton class. Assuming that's correct, why do we want to have separate instances of `@registry` and `@lazy_constants` for each language lexer?

Reply:
Since these aren't within methods, they're defined on the singleton class. `@registry` is indexed by tag, `@lazy_constants` by constant name.

Reply:
Well, that is indeed embarrassing. My roots in Java poke through, I think :P